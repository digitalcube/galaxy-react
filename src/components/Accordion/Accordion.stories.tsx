import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Accordion, AccordionProps } from './Accordion';

const meta: Meta = {
  title: 'Components/Accordion',
  component: Accordion,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<AccordionProps> = (args) => (
  <Accordion {...args}>ここに中身が入ります。</Accordion>
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  defaultOpen: false,
  title: 'ここにタイトルが入ります',
};
