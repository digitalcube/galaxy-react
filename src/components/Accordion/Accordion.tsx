import React, { FC, PropsWithChildren, useState, CSSProperties } from 'react';
import { Collapse, Button } from 'reactstrap';
import { FaAngleDown, FaAngleUp } from 'react-icons/fa';

export type AccordionProps = PropsWithChildren<{
  defaultOpen: boolean;
  id: string;
  title: string;
  style?: CSSProperties;
}>;

export const Accordion: FC<AccordionProps> = ({
  defaultOpen = false,
  children,
  id,
  style,
  title,
}) => {
  const [isOpen, setIsOpen] = useState(defaultOpen);
  const toggle = () => setIsOpen(!isOpen);

  return (
    <div className="accordion border rounded px-3 py-2" style={style}>
      <Button
        id={id}
        onClick={toggle}
        defaultOpen={false}
        className="d-flex justify-content-between align-items-center py-0 px-0 bg-transparent border-0"
      >
        <span className="accordion-title font-weight-bold text-body">
          {title}
        </span>
        <span
          className="h4 mb-0"
          style={{ color: 'Var(--primary-link-hover)' }}
        >
          {!isOpen ? (
            <FaAngleDown className="d-block" />
          ) : (
            <FaAngleUp className="d-block" />
          )}
        </span>
      </Button>
      <Collapse className="accordion-body" isOpen={isOpen}>
        {children}
      </Collapse>
    </div>
  );
};
