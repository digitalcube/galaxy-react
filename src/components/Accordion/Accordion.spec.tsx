import { render } from '@testing-library/react';
import React from 'react';
import { Accordion } from './Accordion';
import { Default as AccordionStory } from './Accordion.stories';

describe('Accordion', () => {
  it('to match snapshot(default)', () => {
    const { asFragment } = render(
      <AccordionStory {...(AccordionStory.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('if defaultOpen', () => {
    const { container } = render(
      <Accordion
        defaultOpen={true}
        id={'test'}
        title={'ここにタイトルが入ります。'}
      />
    );
    expect(container.firstChild).not.toEqual(null);
  });
});
