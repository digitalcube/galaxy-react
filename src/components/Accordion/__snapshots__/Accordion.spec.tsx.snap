// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion to match snapshot(default) 1`] = `
<DocumentFragment>
  <div
    class="accordion border rounded px-3 py-1 accordion-open"
  >
    <button
      class="d-flex justify-content-between align-items-center py-0 px-0 bg-transparent border-0 btn btn-secondary"
      type="button"
    >
      <span
        class="accordion-title font-weight-bold text-body"
      >
        ここにタイトルが入ります
      </span>
      <span
        class="h4 mb-0"
      >
        <svg
          class="d-block"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          />
        </svg>
      </span>
    </button>
    <div
      class="accordion-body collapse"
    >
      ここに中身が入ります。
    </div>
  </div>
</DocumentFragment>
`;
